ptype/full_rap_apply.py

```diff
@@ -0,0 +1,167 @@
+import yaml, glob
+import pandas as pd
+from sklearn.model_selection import train_test_split, GroupShuffleSplit
+from sklearn.preprocessing import StandardScaler
+import tqdm
+import numpy as np
+import tensorflow as tf
+import matplotlib.pyplot as plt
+import matplotlib.lines as mlines
+from matplotlib import cm
+import utils
+import time
+import random
+
+import xarray as xr
+
+from cartopy import crs as ccrs
+from cartopy import feature as cfeature
+import imageio
+from PIL import Image
+from pathlib import Path
+from datetime import datetime, timedelta
+import xarray as xr
+from scipy.ndimage.filters import gaussian_filter
+
+import matplotlib.colors
+
+# Code based on https://unidata.github.io/MetPy/latest/examples/Four_Panel_Map.html 
+def plot_background(ax, bbox):
+    res = "50m"
+    ax.set_extent([bbox[2], bbox[3], bbox[1], bbox[0]])
+    ax.add_feature(cfeature.LAND.with_scale(res))
+    ax.add_feature(cfeature.OCEAN.with_scale(res))
+    ax.add_feature(cfeature.COASTLINE.with_scale(res))
+    ax.add_feature(cfeature.LAKES.with_scale(res), alpha=0.5)
+    ax.add_feature(cfeature.STATES.with_scale(res))
+    return ax
+
+def full_rap_apply(date, time, model_loc, means, stdevs, preds_save_loc, map_save_loc):
```

Comment:
Can you split this function into two functions that separately handle full grid prediction and plotting?